src/Xamarin.Android.Build.Tasks/Tasks/SetMSBuildExtensionsPath.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+
+namespace Xamarin.Android.Tasks
+{
+	public sealed class SetMSBuildExtensionsPath : Task
+	{
+		const   string      MSBuildExtensionsPath           = "MSBuildExtensionsPath";
+		const   string      XBUILD_FRAMEWORK_FOLDERS_PATH   = "XBUILD_FRAMEWORK_FOLDERS_PATH";
+		public override bool Execute ()
+		{
+			var frameworksPath  = Path.GetDirectoryName (typeof (SetMSBuildExtensionsPath).Assembly.Location);
```

Comment:
Where is this task being loaded from? Can this path that we are trying to determine here, be computed in the targets and passed as input here?